pkg/devfile/parser/util/utils.go

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2022-2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/devfile/library/v2/pkg/util"
+	"github.com/hashicorp/go-multierror"
+)
+
+// Default filenames for create devfile
+const (
+	OutputDevfileYamlPath = "devfile.yaml"
+)
+
+type DevfileUtilsClient struct {
+}
+
+func NewDevfileUtilsClient() DevfileUtilsClient {
+	return DevfileUtilsClient{}
+}
+
+// DownloadInMemory is a wrapper to the util.DownloadInMemory() call.
+// This is done to help devfile/library clients invoke this function with a client.
+func (c DevfileUtilsClient) DownloadInMemory(params util.HTTPRequestParams) ([]byte, error) {
+	return util.DownloadInMemory(params)
+}
+
+// DownloadGitRepoResources downloads the git repository resources
+func (c DevfileUtilsClient) DownloadGitRepoResources(url string, destDir string, token string) error {
+	var returnedErr error
+	if util.IsGitProviderRepo(url) {
+		gitUrl, err := util.NewGitURL(url, token)
+		if err != nil {
+			return err
+		}
+
+		if !gitUrl.IsFile || gitUrl.Revision == "" || !strings.Contains(gitUrl.Path, OutputDevfileYamlPath) {
```

Comment:
I couldn't recall. but why the giturl path has to contain `odo-devfile.yaml`?

Reply:
should it be `devfile.yaml` or `devfile.yml`?

Reply:
I am not sure, this is a just a cut-paste from  another file when I was re-arranging it

Reply:
I opened a bug to handle this https://github.com/devfile/api/issues/1362